azurerm/resource_arm_storage_blob.go

```diff
@@ -38,113 +42,81 @@ func resourceArmStorageBlob() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+
 			"resource_group_name": resourceGroupNameSchema(),
+
 			"storage_account_name": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
+
 			"storage_container_name": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
+
 			"type": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				ValidateFunc: validateArmStorageBlobType,
+				ValidateFunc: validation.StringInSlice([]string{"block", "page"}, true),
 			},
+
 			"size": {
 				Type:         schema.TypeInt,
 				Optional:     true,
 				ForceNew:     true,
 				Default:      0,
-				ValidateFunc: validateArmStorageBlobSize,
+				ValidateFunc: validate.IntDivisibleBy(512),
 			},
+
 			"content_type": {
 				Type:          schema.TypeString,
 				Optional:      true,
 				Default:       "application/octet-stream",
 				ConflictsWith: []string{"source_uri"},
 			},
+
 			"source": {
 				Type:          schema.TypeString,
 				Optional:      true,
 				ForceNew:      true,
 				ConflictsWith: []string{"source_uri"},
 			},
+
 			"source_uri": {
 				Type:          schema.TypeString,
 				Optional:      true,
 				ForceNew:      true,
 				ConflictsWith: []string{"source"},
 			},
+
 			"url": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
 			"parallelism": {
 				Type:         schema.TypeInt,
 				Optional:     true,
 				Default:      8,
 				ForceNew:     true,
-				ValidateFunc: validateArmStorageBlobParallelism,
+				ValidateFunc: validation.IntAtLeast(0),
 			},
+
 			"attempts": {
 				Type:         schema.TypeInt,
 				Optional:     true,
 				Default:      1,
 				ForceNew:     true,
-				ValidateFunc: validateArmStorageBlobAttempts,
+				ValidateFunc: validation.IntAtLeast(0),
```

Comment:
The same comment as before:

```suggestion
				ValidateFunc: validation.IntAtLeast(1),
```